consensus/state.go

```diff
@@ -837,11 +1139,12 @@ func (cs *ConsensusState) FinalizeCommit(height int) {
 	log.Info(Fmt("Finalizing commit of block: %v", cs.ProposalBlock))
 	// We have the block, so stage/save/commit-vote.
 	cs.saveBlock(cs.ProposalBlock, cs.ProposalBlockParts, cs.Votes.Precommits(cs.CommitRound))
-	// Increment height.
-	cs.updateToState(cs.stagedState)
+
+	// call updateToState from handleTimeout
```

Comment:
Ok. The need to do this was motivated by writing the test for a single validator: I wanted to get the prevote event, the precommit event, check that they are both in the vote_set for that height, and then let the commit happen. Problem is, it rolls all the way through to the next height before passing through receiveRoutine again, so we don't have any external control over it. So by the time I can check the vote_set for my test's assertion, its already at the next height. 

Maybe its a red herring, and the tests shouldnt be trying to match events to the internal round state? but regardless, I would think we'd want to be able to eg stop and inspect the state before it rolls over into the new height